frontend/src/registration/actions.ts

```diff
@@ -2,58 +2,131 @@ import { push } from 'connected-react-router';
 import { OcimThunkAction } from 'global/types';
 import { Action } from 'redux';
 import { Types as UIActionTypes } from 'ui/actions';
-import { RegistrationModel } from './models';
+import {
+  RegistrationModel,
+  RegistrationStateModel,
+  RegistrationFeedbackModel
+} from './models';
 
 export enum Types {
+  // Support action to update root state
+  ROOT_STATE_UPDATE = 'ROOT_STATE_UPDATE',
+  // Data validation actions
+  REGISTRATION_VALIDATION = 'REGISTRATION_VALIDATION',
+  REGISTRATION_VALIDATION_SUCCESS = 'REGISTRATION_VALIDATION_SUCCESS',
+  REGISTRATION_VALIDATION_FAILURE = 'REGISTRATION_VALIDATION_FAILURE',
+  // Account registration
   REGISTRATION_SUBMIT = 'REGISTRATION_SUBMIT',
   REGISTRATION_SUCCESS = 'REGISTRATION_SUCCESS',
   REGISTRATION_FAILURE = 'REGISTRATION_FAILURE'
 }
 
+export interface RootStateUpdate extends Action {
+  readonly type: Types.ROOT_STATE_UPDATE;
+  readonly data: RegistrationStateModel;
+}
+
+export interface RegistrationValidation extends Action {
+  readonly type: Types.REGISTRATION_VALIDATION;
+}
+
+export interface RegistrationValidationSuccess extends Action {
+  readonly type: Types.REGISTRATION_VALIDATION_SUCCESS;
+  readonly data: RegistrationStateModel;
+}
+
+export interface RegistrationValidationFailure extends Action {
+  readonly type: Types.REGISTRATION_VALIDATION_FAILURE;
+  readonly error: RegistrationFeedbackModel;
+}
+
 export interface SubmitRegistration extends Action {
   readonly type: Types.REGISTRATION_SUBMIT;
-  readonly data: RegistrationModel;
 }
 
 export interface RegistrationSuccess extends Action {
   readonly type: Types.REGISTRATION_SUCCESS;
-  readonly data: RegistrationModel;
+  readonly data: RegistrationStateModel;
 }
 
 export interface RegistrationFailure extends Action {
   readonly type: Types.REGISTRATION_FAILURE;
-  readonly error: any;
+  readonly error: RegistrationFeedbackModel;
 }
 
 export type ActionTypes =
+  | RootStateUpdate
+  | RegistrationValidation
+  | RegistrationValidationSuccess
+  | RegistrationValidationFailure
   | SubmitRegistration
   | RegistrationSuccess
   | RegistrationFailure;
 
+export const updateRootState = (
+  data: RegistrationStateModel
+): OcimThunkAction<void> => async dispatch => {
+  dispatch({ type: Types.ROOT_STATE_UPDATE, data });
+};
```

Comment:
This need not be a ThunkAction since it's not calling any APIs. 
A simplified system can be used like the following:
```javascript
import { createAction } from 'redux-actions';
export const updateRegistration = createAction(Types.ROOT_STATE_UPDATE);
```
Here though it would be better to have:
```javascript
export const updateRegistration = (field: keyof RegistrationStateModel, value) => async dispatch => 
dispatch({ type: Types.REGISTRATION_UPDATE, field, value });
```